### PR TITLE
[stable/redis] Move metrics from deployment to sidecar

### LINKED ISSUE #15658: [stable/redis] Exporter REDIS_ADDR invalid separator in cluster mode
**Describe the bug**
Issues with metrics exporter when deploying a Redis chart (v8.1.2) with cluster and metrics enabled. Exporter container logs errors but it does not crash. As a result, metrics are not published.

**Version of Helm and Kubernetes**:
Helm v2.12.2
Kubernetes v1.12.6

**Which chart**:
stable/redis

**What happened**:
In chart upgrade to 8.0.18, metrics image tag was changed to`v1.0.3`. if you have cluster enabled, it attempts to put in a comma-separated value for the REDIS_ADDR val which is not an acceptable argument.

```
        env:
        - name: REDIS_ADDR
          value: "redis-app-master:6379,redis-app-slave:6379"
```

Logs from redis_exporter
```
time="2019-07-17T18:29:25Z" level=debug msg="Trying DialURL(): redis://redis-app-master:6379,redis-app-slave:6379"
time="2019-07-17T18:29:25Z" level=debug msg="DialURL() failed, err: dial tcp: lookup redis-app-master:6379,redis-app-slave:6379: no such host"
time="2019-07-17T18:29:25Z" level=debug msg="Trying: Dial(): tcp redis-app-master:6379,redis-app-slave:6379"
time="2019-07-17T18:29:25Z" level=error msg="Couldn't connect to redis instance"
time="2019-07-17T18:29:25Z" level=debug msg="connectToRedis( redis-app-master:6379,redis-app-slave:6379 ) err: dial tcp: address redis-app-master:6379,redis-app-slave:6379: too many colons in address"
```

**What you expected to happen**:
It should set up the addresses properly with the supplied separator.

**How to reproduce it** (as minimally and precisely as possible):

- Use chart version 8.1.2
- Set cluster to enabled
- Set metrics to enabled, set debug flag as show below
```
  metrics:
    enabled: true
    extraArgs: {
      "debug": "true"
    }
```
- Deploy chart
- Observe metrics logs and error mentioned above

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier Salmeron Garcia <jsalmeron@vmware.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes https://github.com/helm/charts/issues/15658

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by desaintmartin** [comment]
What about the metrics service ?

**Comment by desaintmartin** [comment]
/assign 
I made a draft pr weeks ago about this but never completed it, nice.

**Review by tompizmor — APPROVED** [review]
/ok-to-test

**Comment by javsalgar** [comment]
@desaintmartin Prometheus will access the pods directly, so we do not need in principle a metrics service

**Comment by javsalgar** [comment]
We just released a Bitnami maintained build of the oliver006/redis_exporter container. As it is a major version, I think it is worth including it in this PR

**Comment by desaintmartin** [comment]
Why using a bitnami-based image? Is the "official" image lagging behind?

**Comment by javsalgar** [comment]
The reason for moving to a bitnami-based image is for us to have full control of the chart containers, so we can have them up-to-date as soon as there is a new version or a security issue.

**Comment by javsalgar** [comment]
I restored the metrics service so it can be used by the ServiceMonitor object that our chart has

File: stable/redis/ci/production-sentinel-values.yaml [hunk 1]
```diff
@@ -420,8 +420,8 @@ metrics:
 
   image:
     registry: docker.io
-    repository: oliver006/redis_exporter
-    tag: v1.0.3
+    repository: bitnami/redis-exporter
+    tag: 1.0.3-r0
```
Comment (andresbono) [inline stable/redis/ci/production-sentinel-values.yaml]: Use `1.0.3-debian-9-r0`

File: stable/redis/ci/production-values.yaml [hunk 2]
```diff
@@ -420,8 +420,8 @@ metrics:
 
   image:
     registry: docker.io
-    repository: oliver006/redis_exporter
-    tag: v1.0.3
+    repository: bitnami/redis-exporter
+    tag: 1.0.3-r0
```
Comment (andresbono) [inline stable/redis/ci/production-values.yaml]: `1.0.3-debian-9-r0`

**Pull request merged**